src/infi/systray/traybar.py

```diff
@@ -250,6 +277,9 @@ def _command(self, hwnd, msg, wparam, lparam):
     def _execute_menu_option(self, id):
         menu_action = self._menu_actions_by_id[id]
         if menu_action == SysTrayIcon.QUIT:
+            nid = NotifyData(self._hwnd, 0)
+            Shell_NotifyIcon(NIM_DELETE, ctypes.byref(nid))
+            PostQuitMessage(0)  # Terminate the app.

```

Comment:
Since this code is now quite similar to the one in `_destroy`, maybe it would be a good idea to have yet another internal function with this code?